cocos/core/utils/js.ts

```diff
@@ -61,18 +60,21 @@ import {
     value,
 } from './js-typed';
 import Pool from './pool';
-
+import * as array from './array';
 import { legacyCC } from '../global-exports';
 
 export * from './js-typed';
 export { default as IDGenerator } from './id-generator';
 export { default as Pool } from './pool';
-export const array = jsarray;
+export { array };
 
+/**
+ * @deprecated all functions expect deprecated function are exported by core.js
+ */
```

Comment:
```suggestion
/**
 * @deprecated since v3.7.0, `js.js` is deprecated, please access `js` directly instead.
 */
```

Reply:
please add 
```
markAsWarning(js, 'js', [
    {
        name: 'js',
        suggest: `'js.js' is deprecated since v3.7.0, please access 'js' directly instead.`,
    },
]);
```
in a new script called `deprecated-3.7.0.ts` 
and import  `deprecated-3.7.0.ts`  in core/index.ts

this prints deprecated warning at engine running time

Reply:
There is already a deprecated js file, why need to add a new one for new version? As not any deprecated API can be removed since v3.x.

Reply:
> There is already a deprecated js file, why need to add a new one for new version? 

the versioned deprecate file is for deprecate module culling by version range